view/model/index.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  Scene,
+  PerspectiveCamera,
+  WebGLRenderer,
+  MeshBasicMaterial,
+  BufferAttribute,
+  BufferGeometry,
+  Vector3,
+  Points,
+  Camera
+} from 'three'
+import { Rgb, P3, Rec2020 } from 'culori'
+import { TrackballControls } from 'three/examples/jsm/controls/TrackballControls'
+
+import { oklch, lch } from '../../lib/colors.js'
+import { showP3, showRec2020 } from '../../stores/settings.js'
+
+interface ModelData {
+  coordinates: Vector3[]
+  colors: Rgb['r'][] | Rgb['g'][] | Rgb['b'][]
+}
+
+let scene: Scene
+let camera: Camera
+let renderer: WebGLRenderer
+let controls: TrackballControls
+
+let canvas = document.querySelector<HTMLCanvasElement>('.model_canvas')!
+
+function init(): void {
+  scene = new Scene()
+  camera = new PerspectiveCamera(75, 3, 0.1, 1000)
+  renderer = new WebGLRenderer({
+    canvas
+  })
+
+  renderer.setPixelRatio(window.devicePixelRatio)
+  renderer.setSize(350, 210)
+  renderer.setClearColor(0xffffff, 0)
+  LCH ? camera.position.setZ(50) : camera.position.setZ(1)
+  LCH ? camera.position.setX(300) : camera.position.setX(2)
+  LCH ? camera.position.setY(200) : camera.position.setY(1)
+  controls = new TrackballControls(camera, renderer.domElement)
+  LCH ? (controls.minDistance = 400) : (controls.minDistance = 2)
+  controls.maxDistance = controls.minDistance
+}
+
+function getModelData(): ModelData {
+  let coordinates = []
+  let colors = []
+
+  let mode: Rgb['mode'] | P3['mode'] | Rec2020['mode'] = 'rgb'
+  if (showRec2020.get()) {
+    mode = 'rec2020'
+  } else if (showP3.get()) {
+    mode = 'p3'
+  }
+
+  for (let x = 0.01; x <= 1; x += 0.01) {
+    for (let y = 0.01; y <= 1; y += 0.01) {
+      for (let z = 0.01; z <= 1; z += 0.01) {
+        let rgb: Rgb | P3 | Rec2020 = { mode, r: x, g: y, b: z }
+        let color
+        LCH ? (color = lch(rgb)) : (color = oklch(rgb))
+        if (color.h) {
+          LCH
+            ? coordinates.push(new Vector3(color.l, color.c, color.h))
+            : coordinates.push(new Vector3(color.l, color.c * 2, color.h / 360))
+          colors.push(rgb.r, rgb.g, rgb.b)
+          if (color.c < 0.05) {
+            coordinates.push(
+              new Vector3(color.l, +(color.c * Math.random()), color.h / 360)
```

Comment:
Do we draw the model my dots? It has 2 problems:

1. The model doesn’t look solid
2. We draw many unnecessary 3D objects (dots inside the model)

Maybe it will be better to detect the end of the model and draw a solid polygon there?

Reply:
I can try the polygone approach, the main problem with it would be detecting edges. 
A lot of simple ways to convert a point cloud to a solid mesh are not suitable for our purpose. Some of them are too crude and loose all the specificity of the shape we need (like convex geometry: our shape is not convex) or expensive to compute (e.g. marching cubes)
Points is a kind of hacky approach, but they are good at approximating our target shape, the points ‘inside’ of the model do not end up being unnecessary because they allow to create a more “solid” overall look from camera distance.
A mix of some algorithm with a more individual approach is what we should aim for because we know our model and can see simple shapes on the edges and can detect the outermost vertices.

Reply:
The problem that we are still didn’t reproduce solid surface even on 2x HiDPI screens. And on phones we can find even 3x screens.

Reply:
I have changed how model is generated to make it completely solid. This approach has it's tradeoffs, but a lot can be fixed with few adjustments I'm planning to make.
I have also moved canvas out of a small card to make it easier to see
 Currently object looks like this: 
![image](https://user-images.githubusercontent.com/73133951/218813189-483890a9-73e0-4926-80e2-0aa902292332.png)